src/NeuralNetwork/index.js

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2019 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+/*
+Generic NeuralNetwork class
+*/
+
+import * as tf from '@tensorflow/tfjs';
+
+class NeuralNetwork {
+  /**
+   * Create a Neural Network.
+   * @param {object} options - An object with options.
+   */
+  constructor(options) {
+    // TODO: create the model based on many more options and defaults
+
+    const inputUnits = options.input || 2;
+    const hiddenUnits = Math.floor(inputUnits / 2) + 1;
+    const outputUnits = options.output || 1;
+
+    this.model = tf.sequential();
+    const hidden = tf.layers.dense({
+      units: hiddenUnits,
+      inputShape: [inputUnits],
+      activation: 'sigmoid',
+    });
+    const output = tf.layers.dense({
+      units: outputUnits,
+      activation: 'sigmoid',
+    });
+    this.model.add(hidden);
+    this.model.add(output);
+
+    const LEARNING_RATE = 0.25;
+    const optimizer = tf.train.sgd(LEARNING_RATE);
+
+    this.model.compile({
+      optimizer,
+      loss: 'meanSquaredError',
+      metrics: ['accuracy'],
+    });
+
+    this.training_xs = [];
+    this.training_ys = [];
+  }
+
+  addData(xs, ys) {
+    this.training_xs.push(xs);
+    this.training_ys.push(ys);
+  }
+
+  async train(optionsOrEpochs, callback) {
```

Comment:
This function is a little wonky b/c it calls the callback continuously. . so that is required here the way I've implemented it. We could use `callCallback()` to support a promise for when it is finished training?

Reply:
Maybe we could have two kinds of callbacks: `whileTraining` and `doneTraining`? As I tried to implement supporting promise and callback, it's hard to keep one callback getting called all the time, and use a promise to return results when it's done training. Then I realize that it's clearer to have separate callbacks. They are both optional.

```javascript
// Use both callbacks
neuralNetwork.train(20, whileTraining, doneTraining);

// Use promise
const trainingResults = await neuralNetwork.train(20);
// trainingResults = { status: 'complete' }

// Use whileTraining callback
const trainingResults = await neuralNetwork.train(20, whileTraining);

// User doneTraining callback
neuralNetwork.train(20, null, doneTraining);
```